racket/commands/macro.rkt

```diff
@@ -158,7 +158,9 @@
   (define after-file  (make-temporary-file-with-text after-text))
   (define out (open-output-string))
   (begin0 (parameterize ([current-output-port out])
-            (system (format "diff -U ~a ~a ~a" -U before-file after-file))
+            (if (eq? (system-type) 'windows)
+                (system (format "diff --strip-trailing-cr -U ~a ~a ~a" -U before-file after-file))
```

Comment:
This assumes that all `diff` utilities someone might use on Windows, have such a `--strip-trailing-cr` command-line flag. Is that a safe assumption?

Reply:
Windows 7 has no `diff` at all. I am using cygwin, so lucky to have it. MinGW also has this `diff` with `--strip-trailing-cr`. So I think it is safe.

PS. I don't know windows 10 or windows 11. I think they have not `diff` as well, because windows has other tool, called `fc` or `windiff`.